api/models/photo.py

```diff
@@ -101,6 +103,22 @@ def _generate_captions_im2txt(self,commit=True):
                 'could not generate im2txt captions for image %s' % image_path)
             return False
 
+    def _generate_clip_embeddings(self, commit):
+        image_path = self.image_paths[0]
```

Comment:
You should use thumbnail_big instead. thumbnail_big is always a JPG, and it is uncommon that a thumbnail is missing. The original picture could have been moved or could be in a file format that is not well understood by semantic_search_instance.calculate_clip_embeddings like a raw image or a video file.

Reply:
Oh! I wasn't aware of that, thanks a lot for this minute observation, I have changed it to thumbnail_big.path